activestorage/lib/active_storage/service.rb

```diff
@@ -121,6 +132,19 @@ def headers_for_direct_upload(key, filename:, content_type:, content_length:, ch
       {}
     end
 
+    def public_service?
+      !!@public_service
+    end
+
+    protected
+      def private_url(key, expires_in:, filename:, disposition:, content_type:, **)
+        raise NotImplementedError
+      end
+
+      def public_url(key, **)
+        raise NotImplementedError
+      end
```

Comment:
Making these methods `protected` signals that they can be called by other services of the same type, but we don’t intend for that to happen. Let’s make them private. You can add `#:doc:` to the end of the `def` line to ensure they appear in the API docs.

Reply:
Thanks for catching this. I don't see a big reason to make these methods show up in the API docs, do you have a particular reason for making these visible in the docs?